webui/react/src/utils/user.ts

```diff
@@ -8,3 +9,25 @@ interface UserNameFields {
 export function getDisplayName(user: DetailedUser | User | UserNameFields | undefined): string {
   return user?.displayName || user?.username || 'Unavailable';
 }
+
+export function isUser(userOrGroup: UserOrGroup): string | undefined {
+  const user = userOrGroup as User;
+  return user?.username || user?.displayName;
+}
+
+export function getName(userOrGroup: UserOrGroup): string {
+  const user = userOrGroup as User;
+  const group = userOrGroup as V1Group;
+  return isUser(userOrGroup) ? getDisplayName(user) : group.name ? group.name : '';
+}
+
+export const getIdFromUserOrGroup = (userOrGroup: UserOrGroup): number => {
+  if (isUser(userOrGroup)) {
+    const user = userOrGroup as User;
+    return user.id;
+  }
+  const group = userOrGroup as V1Group;
+
+  // The groupId should always exist
```

Comment:
Do we expect any groups to exist without a groupId?

Reply:
I agree it should always exist. My experience with `V1Role`: backend didn't make any attribute a required type in the proto annotation.  First I made all attributes required... but it was frustrating because when I added/removed v1Role, it expected all fields. So I made only roleId required. So in this case I would make groupId required and rebuild proto and bindings.